xla/hlo/ir/hlo_computation.cc

```diff
@@ -743,7 +743,10 @@ std::vector<HloInstruction*> HloComputation::MakeInstructionPostOrder(
   dfs_stack_scratch.reserve(instruction_count());
 
   for (const auto& instruction : instructions()) {
-    if (instruction->users().empty()) {
+    if (instruction->users().empty() ||
```

Comment:
I would expect that the first condition `instruction->users().empty()` is now redundant with the second?
And I think it would be good to have a comment here, maybe something like this:
// We don't consider users outside any computation as real users. This can happen when creating new instructions for replacement when cloning computations.

Same below.

Reply:
Yes, that seems accurate. Fixing this. Thanks.